### PR TITLE
Feature/upgrade java 21

### PR DESCRIPTION
**Pull request by dtoulcher-ch**

### CODE REVIEW

**Review by bcullerton — COMMENTED** [review]
Looks good to me. The only change to make is that the folder terraform/groups/strike-off-objections can be removed. The pipeline only attempts to read the folder terraform/groups/ecs-service.

File: terraform/groups/ecs-service/locals.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  name_prefix                 = "${local.stack_name}-${var.environment}"
+  global_prefix               = "global-${var.environment}"
+  service_name                = "psc-discrepancies-api"
+  container_port              = "8080"
+  eric_port                   = "10000"
+  docker_repo                 = "psc-discrepancies.api.ch.gov.uk"
+  kms_alias                   = "alias/${var.aws_profile}/environment-services-kms"
+  lb_listener_rule_priority   = 31
+  lb_listener_paths           = ["/psc-discrepancy-reports*"]
+  healthcheck_path            = "/psc-discrepancy-reports/healthcheck" #healthcheck path for psc-discrepancies-api
```
Comment (bcullerton) [inline terraform/groups/ecs-service/locals.tf:13]: Listener and healthcheck paths need updating for strike-off, see the routes.yaml file for the paths required

File: terraform/groups/ecs-service/locals.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  healthcheck_matcher         = "200"
+  vpc_name                    = local.stack_secrets["vpc_name"]
+  s3_config_bucket            = data.vault_generic_secret.shared_s3.data["config_bucket_name"]
+  app_environment_filename    = "psc-discrepancies.api.ch.gov.uk.env"
```
Comment (bcullerton) [inline terraform/groups/ecs-service/locals.tf:17]: env file needs updating

File: terraform/groups/ecs-service/locals.tf [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# Define all hardcoded local variable and local variables looked up from data resources
+locals {
+  stack_name                  = "company-requests" # this must match the stack name the service deploys into
+  name_prefix                 = "${local.stack_name}-${var.environment}"
+  global_prefix               = "global-${var.environment}"
+  service_name                = "psc-discrepancies-api"
+  container_port              = "8080"
+  eric_port                   = "10000"
+  docker_repo                 = "psc-discrepancies.api.ch.gov.uk"
```
Comment (bcullerton) [inline terraform/groups/ecs-service/locals.tf:9]: docker_repo needs updating

File: terraform/groups/ecs-service/locals.tf [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# Define all hardcoded local variable and local variables looked up from data resources
+locals {
+  stack_name                  = "company-requests" # this must match the stack name the service deploys into
+  name_prefix                 = "${local.stack_name}-${var.environment}"
+  global_prefix               = "global-${var.environment}"
+  service_name                = "psc-discrepancies-api"
```
Comment (bcullerton) [inline terraform/groups/ecs-service/locals.tf:6]: needs updating

File: terraform/groups/ecs-service/variables.tf [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  description = "Toggle default global and shared  environment files"
+}
+variable "log_level" {
+  default     = "info"
+  type        = string
+  description = "The log level for services to use: trace, debug, info or error"
+}
+variable "strike_off_objections_api_version" {
+  type        = string
+  description = "The version of the psc-discrepancies.api.ch.gov.uk container to run."
```
Comment (bcullerton) [inline terraform/groups/ecs-service/variables.tf:130]: Description needs an update here

**Review by bcullerton — COMMENTED** [review]
Terraform code review

File: terraform/groups/ecs-service/locals.tf (hunk 2 above)
Comment (dtoulcher-ch) [reply terraform/groups/ecs-service/locals.tf:17]: I've raised a PR to add the new config to the repo.

**Review by bcullerton — APPROVED** [review]
LGTM, I guess merge once the pipelines set?

**Pull request merged**